x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/ExecutionManager.java

```diff
@@ -129,7 +137,8 @@ public Executable assemble(
                     tsExtractor,
                     tbExtractor,
                     itbExtractor,
-                    i == 0 && descending
+                    i == firstPositive && descending,
```

Comment:
I checked the implementation and I think the redundancy is in the TumblingWindow. The original check in particular, practically just checked that whether first criterion was descending. 
The second criterion was always ascending, so  `criteria.get(0).descending() != criteria.get(1).descending()` 
was practically equivalent to `criteria.get(0).descending()`.

With missing events, that check can be replaced with `criteria.get(matcher.firstPositiveStage).descending()`.

The code in ExecutionManager is correct, the first positive criterion is the query that has to be evaluated in descending order in case of TAIL queries; we cannot change it, or we'll lose the asc/desc information